src/main/java/at/hannibal2/skyhanni/features/misc/DraconicSacrificeTracker.kt

```diff
@@ -0,0 +1,140 @@
+     * REGEX-TEST: §c§lSACRIFICE! §r§eYou turned §r§6Wise Dragon Helmet §r§einto §r§d25 Dragon Essence§r§e!
+     */
+    private val sacrificeLoot by patternGroup.pattern(
+        "sacrifice",
+        "§c§lSACRIFICE! §r§eYou turned §r(?<item>.*) §r§einto §r§d(?<amount>\\d+) Dragon Essence§r§e!",
+    )
+
+    /**
+     * REGEX-TEST: §c§lBONUS LOOT! §r§eYou also received §r§817x §r§5Wise Dragon Fragment §r§efrom your sacrifice!
+     */
+    private val bonusLoot by patternGroup.pattern(
+        "bonus",
+        "§c§lBONUS LOOT! §r§eYou also received §r(?:§\\w(?<amount>\\d+)?x)?(?: §r)?(?<item>.*) §r§efrom your sacrifice!",
+    )
+
+    private val tracker =
+        SkyHanniItemTracker("Draconic Sacrifice Profit Tracker", { Data() }, { it.draconicSacrificeTracker }) {
+            drawDisplay(it)
+        }
+
+    private val altarArea = AxisAlignedBB(-601.0, 4.0, -282.0, -586.0, 15.0, -269.0)
+
+    class Data : ItemTrackerData() {
+        override fun resetItems() {
+            sacrificedItemsMap.clear()
+            itemsSacrificed = 0
+        }
+
+        override fun getDescription(timesGained: Long): List<String> {
+            val percentage = timesGained.toDouble() / itemsSacrificed
+            val dropRate = LorenzUtils.formatPercentage(percentage.coerceAtMost(1.0))
+            return listOf(
+                "§7Dropped §e${timesGained.addSeparators()} §7times.",
+                "§7Your drop rate: §c$dropRate.",
+            )
+        }
+
+        override fun getCoinName(item: TrackedItem) = "§dDragon Essence"
+
+        override fun getCoinDescription(item: TrackedItem): List<String> {
+            val essences = item.totalAmount.addSeparators()
+            return listOf(
+                "§7Sacrificed items give you dragon essence.",
+                "§7You got §6$essences essence §7that way.",
+            )
+        }
+
+        @Expose
+        var itemsSacrificed = 0L
+
+        @Expose
+        var sacrificedItemsMap: MutableMap<String, Long> = mutableMapOf()
+    }
+
+    private fun drawDisplay(data: Data): List<Searchable> = buildList {
+        addSearchString("§5§lDraconic Sacrifice Profit Tracker")
+        val profit = tracker.drawItems(data, { true }, this)
+
+        add(
+            Renderable.hoverTips(
+                "§b${data.itemsSacrificed.addSeparators()} §6Items Sacrified",
+                data.sacrificedItemsMap.map { (key, value) -> "$key: §b$value" },
+            ).toSearchable()
+        )
+
+        add(tracker.addTotalProfit(profit, data.itemsSacrificed, "sacrifice"))
+
+        tracker.addPriceFromButton(this)
+    }
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        sacrificeLoot.matchMatcher(event.message) {
+            val amount = group("amount").toInt()
+            val item = group("item")
+            tracker.addItem("ESSENCE_DRAGON".asInternalName(), amount, command = false)
```

Comment:
initialize essence_dragon as internal name in the class and not inside the function, so you dont have to initialize this every time